### PR TITLE
Add unsigned int to caml_parse_sign_and_base

### LINKED ISSUE #790: Scanf.sscanf failing with %u formatter
```ocaml
Scanf.sscanf "0.97.0" "%u.%u.%u" (fun major minor patch -> [major; minor; patch])
```

This throws with

```
    function bad_input(s){throw [0,Scan_failure,s]}
                          ^
0,248,Stdlib.Scanf.Scan_failure,15,scanf: bad input at char number 1: int_of_string
```

But %i and %d works

### PR DESCRIPTION
**Pull request by goodmind**
Fixes #790

### CODE REVIEW

**Comment by goodmind** [comment]
@hhugo

**Comment by hhugo** [comment]
merged in https://github.com/ocsigen/js_of_ocaml/pull/793

**Comment by hhugo** [comment]
(Note, I've cherry-picked your commit)

**Comment by hhugo** [comment]
Thanks

File: runtime/stdlib.js [hunk 1]
```diff
@@ -447,6 +447,7 @@ function caml_parse_sign_and_base (s) {
     case 120: case 88: base = 16; i += 2; break;
     case 111: case 79: base =  8; i += 2; break;
     case  98: case 66: base =  2; i += 2; break;
+    case 117: case 85: sign =  0; i += 2; break;
```
Comment (hhugo) [inline runtime/stdlib.js:450]: `sign = 0` was not correct here.

**Pull request closed without merging**